frontend/src/app/components/settings/settings.component.ts

```diff
@@ -117,6 +121,35 @@ export class SettingsComponent implements OnInit {
     });
   }
 
+  confirmDelete(): void {
+    const data = {
+      title: 'Confirm Delete',
+      body: 'Are you sure you want to delete your account? This action cannot be undone',
+      cancelText: 'CANCEL',
+      confirmText: 'CONFIRM',
+    };
+
+    this.dialogService.open(data.title, data.body, data.cancelText, data.confirmText);
```

Comment:
if you find it more conveient to construct the parameters as an object (which imo is very reasonable), then it might also be a good idea to have `dialogService` take that one object as its only parameter rather than as separate strings. this might also be helpful at making sure that callers never provide the strings in the wrong order.

Reply:
js doesn't need key word args! /s